include/emp/Evolve/Systematics.hpp

```diff
@@ -416,9 +464,13 @@ namespace emp {
     using hash_t = typename Ptr<taxon_t>::hash_t;
     using fun_calc_info_t = std::function<ORG_INFO(ORG &)>;
 
-    fun_calc_info_t calc_info_fun;
-    Ptr<taxon_t> next_parent;
-    Ptr<taxon_t> most_recent;
+    fun_calc_info_t calc_info_fun; ///< Function that takes an organism and returns the unit being tracked by systematics
+    Ptr<taxon_t> next_parent;      ///< The taxon that has been marked as parent for next new org
+    Ptr<taxon_t> most_recent;      ///< The most-recently added taxon
+    bool num_orgs_wrong = false;   ///< Keep track of whether we have loaded from a file that didn't
```

Comment:
Hmm, I'm conflicted. I like the safety/clarity that optional would provide. However, it looks like it's a not insubstantial time/space hit for numbers that are used in every taxon and operated on a fair amount. It will also make the code potentially confusing to folks who are not familiar with std::optional. Particularly because it's actually very expedient to assume that trees loaded in without org counts have one of each org. It makes all of the topology stuff and tree modification operations work fine. It just makes a couple of metrics inaccurate.

Reply:
Ah, so when `num_orgs_wrong` is `true`, `num_orgs` is still being used but just at a defaulted value? Maybe name these variables `_defaulted` instead of `_wrong`?

Reply:
It should compile down to the same thing https://godbolt.org/z/e8hs4G1bG To save space you'd have to do something like pack the two bools into a flag field.

Reply:
I'm happy to switch to "defaulted" (although that kind of feels like a euphemism for wrong :sweat_smile:) if you think that would be clearer.

I could be wrong here, but isn't part of the reason it's compiling down to the same thing on godbolt the fact that in that case the compiler knows everything ahead of time and so can optimize out branching introduced by std::optional?

Reply:
Example: https://godbolt.org/z/W78vj19Pz

Reply:
And here's a quick benchmark (this is actually a bigger effect than I expected): https://quick-bench.com/q/8tBRUrz5poxkHfQ27uDBC8BWgFE

Reply:
As long as you deref it it's the same https://godbolt.org/z/v6MMjcYaW

Reply:
https://quick-bench.com/q/5yjQDOiRas0cHzoqrECt4cytP7A

Reply:
`    if (*optional.foo > 5) {`

Reply:
I think the big difference has to do with passing the optional as a param --- semantically different from just passing an int

Reply:
My interpretation of `_defaulted` would be "it might not be right but it's reasonable" vs `_wrong` would be "all bets are off"

Reply:
That''s what I was guessing (which I think actually would be concerning, because if we're using optionals we're going to have be passing them around, etc. - it's semantically different, but representative of the scenarios we're comparing), but then I tested it out and it looks like it is really just the deref that makes the difference

Reply:
Okay, that's fair for `_defaulted`.

Reply:
Either way could be a good call!

Reply:
I was almost convinced, but here's the problem. The dereference operator is undefined behavior if there's not actually a value in the optional object. So then you end up needing to use `.value()`. Which A) appears to be back to the slow speed (https://quick-bench.com/q/acv1zCs2Agso3yZKhfBolRQ8rx0) and B) potentially throws an exception, which we would love to avoid ever having happen in code compiled via emscripten.

Reply:
I'm going to go ahead and just used `_defaulted`, because the cases where we care that its sometimes wrong are way rarer than the cases where we don't. If that balance shifts we can reconsider.